pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/DispatchablePlanMetadata.java

```diff
@@ -80,6 +81,24 @@ public class DispatchablePlanMetadata implements Serializable {
   // Map from workerId -> {planFragmentId -> mailboxes}
   private final Map<Integer, Map<Integer, MailboxInfos>> _workerIdToMailboxesMap = new HashMap<>();
 
+  /**
+   * Map from workerId -> {tableType -> {tableName -> segments}} is required for logical tables.
+   * Raw definition of the map is:
+   * Map<String, Map<String, Map<String, List<String>>>>. Since this definition is hard to understand - specifically
+   * what do each of the string keys store, we define two classes:
+   * {@link TableTypeToSegmentsMap} and {@link TableTypeTableNameToSegmentsMap} to help read code more easily.
+   */
+  public static class TableTypeToSegmentsMap {
```

Comment:
I'd suggest just keeping the plain map. Each class introduced is adding overhead on heap. Putting some javadoc explaining what each level stands for should be good enough.

Reply:
Some of the map can be optimized into Pairs to reduce overhead

Reply:
I wish Java had a typedef like language construct. I can change it now since the code works. FYI: this data structure caused the most amount of delay in making this feature work.

Reply:
Pair can work in general. However serialization, deserialization using jackson is proving to be difficult. https://github.com/FasterXML/jackson-databind/issues/1156

Reply:
Doh! - Simplified the data structure to `Map<String, List<String>>`. I realised that the data structure has physical table name AND table type. So now it is a map of `physicalTableName -> segments`. 

I did this AFTER spending a lot of time thinking about jackson map serialization and other related issues.